### PR TITLE
c 0.14

### PR DESCRIPTION
**Pull request by Freed-Wu**
Copied from <https://github.com/ryanmjacobs/c/blob/master/c.rb>

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/c.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+  license "MIT"
+  head "https://github.com/ryanmjacobs/c.git", branch: "master"
+
+  def install
+    bin.install "c"
+  end
+
+  test do
+    assert_match "Execute C programs from the command line.", \
+      shell_output("#{bin}/c 2>&1", 1)
```
Comment (carlocab) [inline Formula/c.rb:15]: Can you propose a more substantial test than checking the version or calling for help? These tests rarely ever fail even for broken installations.

>We want tests that don’t require any user input and test the basic functionality of the application. For example `foo build-foo input.foo` is a good test and (despite their widespread use) `foo --version` and `foo --help` are bad tests. 

Ref. https://docs.brew.sh/Formula-Cookbook#add-a-test-to-the-formula

**Review by carlocab — CHANGES_REQUESTED** [review]
Please use the preferred commit-message style for homebrew/core. We put the name of the formula first in commit-message headings.

For new formulae:

> At Homebrew, we like to put the name of the formula up front like so: `foobar 7.3 (new formula)`.

For existing formulae:

> The preferred commit message format for simple version updates is `foobar 7.3` and for fixes is `foobar: fix flibble matrix.`.

Refer to the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit) for more details. Also, when making further changes to your pull request, use the following guidelines to make sure that @BrewTestBot can merge your commits:

> - One formula per commit; one commit per formula.
> - Keep merge commits out of the pull request.

File: Formula/c.rb (hunk 1 above)
Comment (Freed-Wu) [reply Formula/c.rb]: OKay.

File: Formula/c.rb [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+  license "MIT"
+  head "https://github.com/ryanmjacobs/c.git", branch: "master"
+
+  def install
+    bin.install "c"
+  end
+
+  test do
+    (testpath/"test.c").write("int main(void){return 0;}")
+    system "#{bin}/c", testpath/"test.c"
```
Comment (carlocab) [inline Formula/c.rb:15]:
```suggestion
    system bin/"c", testpath/"test.c"
```

Comment (Freed-Wu) [reply Formula/c.rb]: Cool!

**Review by carlocab — APPROVED** [review]
Thanks!

**Comment by Freed-Wu** [comment]
You are welcome.

**Review by chenrui333 — APPROVED** [review]

**Comment by chenrui333** [comment]
@Freed-Wu, thanks for your contribution to Homebrew! :tada: 🥇

Without awesome contributors like you, it would be impossible to maintain Homebrew to the high level of quality users have come to expect. Thank you!!!!

**Comment by Freed-Wu** [comment]
You are welcom.

BTW:
<https://github.com/Homebrew/homebrew-core/blob/master/Formula/ripgrep.rb#L43>:

```ruby
    system "#{bin}/rg", "Hello World!", testpath
```

should be

```ruby
    system bin/"rg", "Hello World!", testpath
```

Like this PR, I think.

And any response for #108038 #108039 #109664? All checks have passed for them.

**Pull request closed without merging**